clang/lib/Serialization/ASTReader.cpp

```diff
@@ -7944,9 +7944,13 @@ void ASTReader::PrintStats() {
   std::fprintf(stderr, "*** AST File Statistics:\n");
 
   unsigned NumTypesLoaded =
-      TypesLoaded.size() - llvm::count(TypesLoaded, QualType());
+      TypesLoaded.size() - std::count(TypesLoaded.materialisedBegin(),
```

Comment:
This seems confusing - I'd have thought any materialised type was a loaded type? So `NumTypesLoaded = std::count(TypesLoaded.materialisedBegin(), TypesLoaded.materialisedEnd())`?

(similarly for `DeclsLoaded` below)

Like wouldn't this result in `NumTypesLoaded` too high - for unmaterialized values they're all assumed to be loaded? Which sounds backwards.

Reply:
No, materialised simply means the page holding it was allocated. One still needs to look for non default initialised elements as before.